perun-base/src/test/resources/sponsored-accounts-config.yml

```
@@ -0,0 +1,8 @@
+---
+namespaces:
+
+  dummy_namespace:
+    attributes_rules:
+      login: REQUIRED
+      password: OPTIONAL
```

I think it would make it easier to work with if the format was a bit different:
```suggestion
  dummy_namespace:
    required_attributes:
    - login
    optional_attributes:
    - password
```
Also, in the NamespaceRules class, it would be nice to have two Set<String> of attributes, one for optional and one for required attributes. Using just a Map<String, String> makes it much harder to decode the actual structure of the data. @balcirakpeter what do you think?

---

Reworked